cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -973,22 +972,33 @@ toAlonzoPParamsUpdate
     , Alonzo._minPoolCost     = toShelleyLovelace <$>
                                   maybeToStrictMaybe protocolUpdateMinPoolCost
     , Alonzo._coinsPerUTxOWord  = toShelleyLovelace <$>
-                                  maybeToStrictMaybe protocolUpdateUTxOCostPerWord
+                                  maybeToStrictMaybe
+                                    (protocolUpdateUTxOCostPerWord >>= non 0)
     , Alonzo._costmdls        = if Map.null protocolUpdateCostModels
                                   then Ledger.SNothing
                                   else Ledger.SJust
                                          (toAlonzoCostModels protocolUpdateCostModels)
     , Alonzo._prices          = toAlonzoPrices  <$>
-                                  maybeToStrictMaybe protocolUpdatePrices
+                                  maybeToStrictMaybe
+                                    (protocolUpdatePrices
+                                      >>= non (ExecutionUnitPrices 0 0))
     , Alonzo._maxTxExUnits    = toAlonzoExUnits  <$>
                                   maybeToStrictMaybe protocolUpdateMaxTxExUnits
     , Alonzo._maxBlockExUnits = toAlonzoExUnits  <$>
                                   maybeToStrictMaybe protocolUpdateMaxBlockExUnits
-    , Alonzo._maxValSize      = maybeToStrictMaybe protocolUpdateMaxValueSize
-    , Alonzo._collateralPercentage = maybeToStrictMaybe protocolUpdateCollateralPercent
-    , Alonzo._maxCollateralInputs  = maybeToStrictMaybe protocolUpdateMaxCollateralInputs
+    , Alonzo._maxValSize      = maybeToStrictMaybe $
+                                protocolUpdateMaxValueSize >>= non 0
+    , Alonzo._collateralPercentage  = maybeToStrictMaybe $
+                                        protocolUpdateCollateralPercent
+                                        >>= non 0
+    , Alonzo._maxCollateralInputs   = maybeToStrictMaybe $
+                                        protocolUpdateMaxCollateralInputs
+                                        >>= non 0
     }
 
+-- | Check if value is not special.
+non :: (Alternative f, Eq a) => a -> a -> f a
+non z n = n <$ guard (n /= z)
```

Comment:
All the uses of this are deeply suspicious. There are not special values like this.

Reply:
Well, it's strange, I found that implementation treats Just 0 and Nothing the same way, so I added some unification here. Now tests say it is wrong. I removed this.